code/modules/mob/living/carbon/carbon.dm

```diff
@@ -79,38 +79,41 @@
 		return shock_damage
 
 
-/mob/living/carbon/swap_hand()
-	var/obj/item/item_in_hand = src.get_active_hand()
+/mob/living/carbon/swap_hand(held_index)
+	if(!held_index)
+		held_index = (held_index % held_items.len)+1

```

Comment:
I do need to look at this bit, I was overzealous so this line probably exists in too many places, it also sometimes fails to go up an index, so it might just be wrong too.